apps/api/src/utils/mapInBatches.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Calls callback `options.batchSize` times in parallel for every item in array sequentially.
+ * This is useful for rate limiting requests to external services.
+ *
+ * @returns an array of mapped results
+ */
+export async function mapInBatches<T, U>(items: T[], callback: (item: T, index: number) => Promise<U>, options?: MapInBatchesOptions): Promise<U[]> {
```

Comment:
question (non-blocking): I fail to see where this is used

Reply:
Yes, nice catch! too much stashes in the late evening